OverlayPlugin.Core/NetworkProcessors/LineAbilityExtra.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace RainbowMage.OverlayPlugin.NetworkProcessors
+{
+    /**
+     * Class for position/angle info from ActionEffect (i.e. line 21/22).
+     *
+     * This one is implemented different than LineActorCast. ActionEffect has several variants with different
+     * sizes to support different maximum numbers of targets. They have a common header, which contains the angle of
+     * the cast. However, the x/y/z coordinates are *after* the variable-size portion, so those are accessed by negative
+     * indexing from the end of the packet rather than reflectively via Machina definitions.
```

Comment:
This is a subjective opinion rather than some sort of hard requirement or anything, but I think I'd prefer to see these mapped out properly as structs to match other network packet logic, as well as to make it more clear exactly where in the packet this data resides rather than leaving it as a nebulous "X bytes from the end".

I think it would be better to have something like:

```cs
[StructLayout(LayoutKind.Sequential, Pack = 1, Size = 0x29C)]
private struct Server_ActionEffect8_Extra
{
    [FieldOffset(0x290)]
    public ushort RawX;

    [FieldOffset(0x292)]
    public ushort RawX;

    [FieldOffset(0x294)]
    public ushort RawX;
}
```

Then a sanity check in the constructor to make sure that our packet size matches FFXIV_ACT_Plugin's packet size while mapping them.

In the actual handler, you can marshal to these types (or just raw cast like e.g. MapEffect or RSV lines do), then you can reference the fields directly.

My biggest concern with the way you're handling it below is that some update will adjust these positions in the packet or add new data at the end, and it won't be obvious what's causing bad data in that case.

Reply:
Done